templates/swift/Sources/OAuth/WebAuthComponent.swift.twig

```diff
@@ -130,5 +148,20 @@ public class WebAuthComponent {
         callbacks.forEach { (_, callback) in
             callback(.failure({{ spec.title | caseUcfirst}}Error(message: "User cancelled login.")))
         }
+
+        #if canImport(AuthenticationServices)
+        currentAuthSession = nil
+        #endif
+    }
+}
+
+#if canImport(AuthenticationServices)
+/// Presentation context for the ASWebAuthenticationSession.
+class PresentationContextProvider: NSObject, ASWebAuthenticationPresentationContextProviding {
+    static let shared = PresentationContextProvider()
+
+    func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+        return ASPresentationAnchor()
     }
```

Comment:
Not sure this will work as expected on macOS, can we anchor to the top-most window instead?

Reply:
Done.